src/main/scala/device/MemEncryptUtil.scala

```diff
@@ -0,0 +1,808 @@
+/***************************************************************************************
+* Copyright (c) 2024-2025 Institute of Information Engineering, Chinese Academy of Sciences
+*
+* XiangShan is licensed under Mulan PSL v2.
+* You can use this software according to the terms and conditions of the Mulan PSL v2.
+* You may obtain a copy of Mulan PSL v2 at:
+*          http://license.coscl.org.cn/MulanPSL2
+*
+* THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND,
+* EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT,
+* MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.
+*
+* See the Mulan PSL v2 for more details.
+***************************************************************************************/
+
+package device
+
+import chisel3._
+import chisel3.util._
+import chisel3.util.HasBlackBoxResource
+import org.chipsalliance.cde.config.Parameters
+import freechips.rocketchip.amba.axi4._
+import freechips.rocketchip.diplomacy._
+import freechips.rocketchip.util._
+import freechips.rocketchip.amba.apb._
+import freechips.rocketchip.tilelink.AXI4TLState
+
+class GetCKI extends Module {
```

Comment:
What is CKI? We need some simplest comments here to help people to have a basic understanding of what work this code is doing.